app/controllers/application_controller.rb

```diff
@@ -207,5 +209,15 @@ def avatar(name, founder: nil, faculty: nil, version: :mid, background_shape: :c
     ).svg.html_safe
   end
 
-  helper_method :avatar
+  def domain_redirection_required?
+    return false if current_domain.blank?
+
+    return false if current_domain.primary? || current_school.domains.one?
+
+    !!current_school.configuration['redirect_to_primary_domain']
```

Comment:
I think the default behavior should be to redirect to the primary domain. 
The school owner could optionally disable redirection by setting the school configuration. 

(In most cases people what the school to be in a single domain name)